### DISCUSSION TITLE
Deploying Open5GS services into separate containers

### DISCUSSION THREAD

**Discussion by MerouaM** [herlesupreeth/docker_open5gs#340] [Q&A]
Hello,

Is there a way to deploy each service of each Open5GS network function into its own docker container?

Thank you in advance :)

**Comment by herlesupreeth** [comment]
This repo is already doing that

**Comment by MerouaM** [comment]
From what I understand, this repository deploys each network function (NF) in a container. However, I want to further split each NF into separate services and deploy each service in its own specific container.

Is this possible?

**Reply by herlesupreeth** [reply]
~I dont think open5gs supports splitting NFs into separate services~

**Reply by MerouaM** [reply]
Are the NFs monolithic? For instance, is it possible to create two instances of the same NF (e.g., AMF), deactivate some services in the first instance, and activate those same services in the second instance?

**Reply by herlesupreeth** [reply]
Correction: I think one can select which services to run in an NF in open5gs using the below example configuration for AMF (I had forgotten about it since this was removed from cfg example in the recent open5gs tag)

```
#
#  o NF Service Name(Default : all NF services available)
#    service_name:
#
#  o NF Service Name(Only some NF services are available)
#    service_name:
#      - namf-comm
#
```

> Are the NFs monolithic?

No

> For instance, is it possible to create two instances of the same NF (e.g., AMF), deactivate some services in the first instance, and activate those same services in the second instance?

yes, possible

> Is there a way to deploy each service of each Open5GS network function into its own docker container?

yes, you can. But, with the current state deploying more than one instance of a NF (e.g. AMF) involves a bit of extra steps. Easing the such deployments is something on my TODO list.

**Reply by MerouaM** [reply]
Thank you so much for this detailed explanation !
I will work on this too! Since I need this for my PoC.
I will keep you updated 😉

**Reply by MerouaM** [reply]
Hello @herlesupreeth 

Where can I find this configuration file ?


  o NF Service Name(Default : all NF services available)
    service_name:

  o NF Service Name(Only some NF services are available)
    service_name:
      - namf-comm


Thank you :)

**Reply by herlesupreeth** [reply]
In the latest tag of open5gs those example configuration have been removed but you can see them in the code

```
#define OGS_SBI_SERVICE_NAME_NNRF_NFM "nnrf-nfm"
#define OGS_SBI_SERVICE_NAME_NNRF_DISC "nnrf-disc"
#define OGS_SBI_SERVICE_NAME_NNRF_OAUTH2 "nnrf-oauth2"
#define OGS_SBI_SERVICE_NAME_NUDM_SDM "nudm-sdm"
#define OGS_SBI_SERVICE_NAME_NUDM_UECM "nudm-uecm"
#define OGS_SBI_SERVICE_NAME_NUDM_UEAU "nudm-ueau"
#define OGS_SBI_SERVICE_NAME_NUDM_EE "nudm-ee"
#define OGS_SBI_SERVICE_NAME_NUDM_PP "nudm-pp"
#define OGS_SBI_SERVICE_NAME_NUDM_NIDDAU "nudm-niddau"
#define OGS_SBI_SERVICE_NAME_NUDM_MT "nudm-mt"
#define OGS_SBI_SERVICE_NAME_NAMF_COMM "namf-comm"
#define OGS_SBI_SERVICE_NAME_NAMF_EVTS "namf-evts"
#define OGS_SBI_SERVICE_NAME_NAMF_MT "namf-mt"
#define OGS_SBI_SERVICE_NAME_NAMF_LOC "namf-loc"
#define OGS_SBI_SERVICE_NAME_NSMF_PDUSESSION "nsmf-pdusession"
#define OGS_SBI_SERVICE_NAME_NSMF_EVENT_EXPOSURE "nsmf-event-exposure"
#define OGS_SBI_SERVICE_NAME_NSMF_NIDD "nsmf-nidd"
#define OGS_SBI_SERVICE_NAME_NAUSF_AUTH "nausf-auth"
#define OGS_SBI_SERVICE_NAME_NAUSF_SORPROTECTION "nausf-sorprotection"
#define OGS_SBI_SERVICE_NAME_NAUSF_UPUPROTECTION "nausf-upuprotection"
#define OGS_SBI_SERVICE_NAME_NNEF_PFDMANAGEMENT "nnef-pfdmanagement"
#define OGS_SBI_SERVICE_NAME_NNEF_SMCONTEXT "nnef-smcontext"
#define OGS_SBI_SERVICE_NAME_NNEF_EVENTEXPOSURE "nnef-eventexposure"
#define OGS_SBI_SERVICE_NAME_NPCF_AM_POLICY_CONTROL "npcf-am-policy-control"
#define OGS_SBI_SERVICE_NAME_NPCF_SMPOLICYCONTROL "npcf-smpolicycontrol"
#define OGS_SBI_SERVICE_NAME_NPCF_POLICYAUTHORIZATION "npcf-policyauthorization"
#define OGS_SBI_SERVICE_NAME_NPCF_BDTPOLICYCONTROL "npcf-bdtpolicycontrol"
#define OGS_SBI_SERVICE_NAME_NPCF_EVENTEXPOSURE "npcf-eventexposure"
#define OGS_SBI_SERVICE_NAME_NPCF_UE_POLICY_CONTROL "npcf-ue-policy-control"
#define OGS_SBI_SERVICE_NAME_NSMSF_SMS "nsmsf-sms"
#define OGS_SBI_SERVICE_NAME_NNSSF_NSSELECTION "nnssf-nsselection"
#define OGS_SBI_SERVICE_NAME_NNSSF_NSSAIAVAILABILITY "nnssf-nssaiavailability"
#define OGS_SBI_SERVICE_NAME_NUDR_DR "nudr-dr"
#define OGS_SBI_SERVICE_NAME_NUDR_GROUP_ID_MAP "nudr-group-id-map"
#define OGS_SBI_SERVICE_NAME_NLMF_LOC "nlmf-loc"
#define OGS_SBI_SERVICE_NAME_N5G_EIR_EIC "n5g-eir-eic"
#define OGS_SBI_SERVICE_NAME_NBSF_MANAGEMENT "nbsf-management"
#define OGS_SBI_SERVICE_NAME_NCHF_SPENDINGLIMITCONTROL \
    "nchf-spendinglimitcontrol"
#define OGS_SBI_SERVICE_NAME_NCHF_CONVERGEDCHARGING "nchf-convergedcharging"
#define OGS_SBI_SERVICE_NAME_NCHF_OFFLINEONLYCHARGING "nchf-offlineonlycharging"
#define OGS_SBI_SERVICE_NAME_NNWDAF_EVENTSSUBSCRIPTION \
    "nnwdaf-eventssubscription"
#define OGS_SBI_SERVICE_NAME_NNWDAF_ANALYTICSINFO "nnwdaf-analyticsinfo"
#define OGS_SBI_SERVICE_NAME_NGMLC_LOC "ngmlc-loc"
#define OGS_SBI_SERVICE_NAME_NUCMF_PROVISIONING "nucmf-provisioning"
#define OGS_SBI_SERVICE_NAME_NUCMF_UECAPABILITYMANAGEMENT \
    "nucmf-uecapabilitymanagement"
#define OGS_SBI_SERVICE_NAME_NHSS_SDM "nhss-sdm"
#define OGS_SBI_SERVICE_NAME_NHSS_UECM "nhss-uecm"
#define OGS_SBI_SERVICE_NAME_NHSS_UEAU "nhss-ueau"
#define OGS_SBI_SERVICE_NAME_NHSS_EE "nhss-ee"
#define OGS_SBI_SERVICE_NAME_NHSS_IMS_SDM "nhss-ims-sdm"
#define OGS_SBI_SERVICE_NAME_NHSS_IMS_UECM "nhss-ims-uecm"
#define OGS_SBI_SERVICE_NAME_NHSS_IMS_UEAU "nhss-ims-ueau"
#define OGS_SBI_SERVICE_NAME_NSEPP_TELESCOPIC "nsepp-telescopic"
#define OGS_SBI_SERVICE_NAME_NSORAF_SOR "nsoraf-sor"
#define OGS_SBI_SERVICE_NAME_NSPAF_SECURED_PACKET "nspaf-secured-packet"
#define OGS_SBI_SERVICE_NAME_NUDSF_DR "nudsf-dr"
#define OGS_SBI_SERVICE_NAME_NNSSAAF_NSSAA "nnssaaf-nssaa"

#define OGS_SBI_SERVICE_NAME_N32C_HANDSHAKE "n32c-handshake"

#define OGS_SBI_SERVICE_NAME_NAMF_CALLBACK "namf-callback"
#define OGS_SBI_SERVICE_NAME_NSMF_CALLBACK "nsmf-callback"
```

**Reply by herlesupreeth** [reply]
https://github.com/open5gs/open5gs/blob/499c70d8be63e5f87c2d1c2bcd4e521d50ce1a6b/configs/open5gs/amf.yaml.in#L179C1-L188C19

**Reply by MerouaM** [reply]
Thank you for your quick response!

I have another question, if you don't mind. Is it possible to deploy two network functions (NFs) offering the same services, and then direct the traffic to one or the other based on the specific service operation? For example, can I deploy two AMFs, both using NAMF-COMM services, where AMF1 handles operations such as UE context management, while AMF2 handles operations like N1N2 messages and notifications?

Also, is there a simpler way to do this ? I mean is it possible to handle this at the configuration step rather than doing packet redirection (if possible in the first place)?

**Reply by herlesupreeth** [reply]
> Is it possible to deploy two network functions (NFs) offering the same services, and then direct the traffic to one or the other based on the specific service operation?

I think when a NF registers with NRF/SCP its mentions which all services it supports (i.e. service_name configured in respective NF) . And whenever a request for discovering such service arrives to NRF/SCP it send the IP of that NF

> For example, can I deploy two AMFs, both using NAMF-COMM services, where AMF1 handles operations such as UE context management, while AMF2 handles operations like N1N2 messages and notifications?

I dont think so

**Reply by MerouaM** [reply]
This means that NFs can be split only on the service level.

Thank you for the explanation :)

**Reply by MerouaM** [reply]
Hello @herlesupreeth !

I am trying to deploy two UDMs and two PCFs, all supporting a sub-group of services:

- udm-1 supports: nudm-sdm, nudm-uecm, nudm-ee, nudm-pp, nudm-niddau, nudm-mt
- udm-2 supports: nudm-ueau, nudm-ee, nudm-pp, nudm-niddau, nudm-mt
- pcf-1 supports: npcf-am-policy-control, npcf-ue-policy-control, npcf-policyauthorization, npcf-bdtpolicycontrol, npcf-eventexposure
- pcf-2 supports: npcf-smpolicycontrol, npcf-policyauthorization, npcf-bdtpolicycontrol, npcf-eventexposure

I have created for every function its own directory with the .sh and yaml files. Here is a part of the directory structure:
- docker_open5gs
          - udm-1
                    - - udm-1.yaml
                    - - udm-1_init.sh
          - udm-2
                    - - udm-2.yaml
                    - - udm-2_init.sh
          - pcf-1
                    - - pcf-1.yaml
                    - - pcf-1_init.sh
          - pcf-2
                    - - pcf-2.yaml
                    - - pcf-2_init.sh
                    
I edited the base/Dockerfile and base/open5gs_init.sh files to suit my setup.
I also edited the sa-deploy.yaml file to suit my setup, I named the new file sa-deploy-2.yaml

As soon as I turn up the NFs, pcf-1 and pcf-2 containers exit with code (0). 

I checked the logs, and I got this:

- PCF-1

`Deploying component: 'pcf-1'
Open5GS daemon v2.7.0-96-g322719f

07/29 18:52:18.222: [app] INFO: Configuration: '/open5gs/install/etc/open5gs/pcf.yaml' (../lib/app/ogs-init.c:130)
07/29 18:52:18.222: [app] INFO: File Logging: '/open5gs/install/var/log/open5gs/pcf.log' (../lib/app/ogs-init.c:133)
07/29 18:52:18.224: [metrics] INFO: metrics_server() [http://127.0.0.13]:9090 (../lib/metrics/prometheus/context.c:299)
07/29 18:52:18.224: [dbi] WARNING: Failed to connect to server [mongodb://localhost/open5gs] (../lib/dbi/ogs-mongoc.c:122)
07/29 18:52:18.224: [app] ERROR: Failed to initialize PCF (../src/pcf/app.c:28)
`

- PCF-2
`Deploying component: 'pcf-2'
Open5GS daemon v2.7.0-96-g322719f

07/29 18:52:18.460: [app] INFO: Configuration: '/open5gs/install/etc/open5gs/pcf.yaml' (../lib/app/ogs-init.c:130)
07/29 18:52:18.460: [app] INFO: File Logging: '/open5gs/install/var/log/open5gs/pcf.log' (../lib/app/ogs-init.c:133)
07/29 18:52:18.462: [metrics] INFO: metrics_server() [http://127.0.0.13]:9090 (../lib/metrics/prometheus/context.c:299)
07/29 18:52:18.463: [dbi] WARNING: Failed to connect to server [mongodb://localhost/open5gs] (../lib/dbi/ogs-mongoc.c:122)
07/29 18:52:18.463: [app] ERROR: Failed to initialize PCF (../src/pcf/app.c:28)
`
The UDM containers stay up, but they cannot connect to the NRF. The logs show the following:

- UDM-1
`Deploying component: 'udm-1'
Open5GS daemon v2.7.0-96-g322719f

07/29 18:52:07.825: [app] INFO: Configuration: '/open5gs/install/etc/open5gs/udm.yaml' (../lib/app/ogs-init.c:130)
07/29 18:52:07.825: [app] INFO: File Logging: '/open5gs/install/var/log/open5gs/udm.log' (../lib/app/ogs-init.c:133)
07/29 18:52:07.827: [sbi] INFO: NF Service [nudm-ueau] (../lib/sbi/context.c:1812)
07/29 18:52:07.827: [sbi] INFO: NF Service [nudm-uecm] (../lib/sbi/context.c:1812)
07/29 18:52:07.827: [sbi] INFO: NF Service [nudm-sdm] (../lib/sbi/context.c:1812)
07/29 18:52:07.827: [sbi] INFO: nghttp2_server() [http://127.0.0.12]:7777 (../lib/sbi/nghttp2-server.c:414)
07/29 18:52:07.827: [app] INFO: UDM initialize...done (../src/udm/app.c:31)
07/29 18:52:07.827: [sbi] WARNING: [7] Failed to connect to 127.0.0.200 port 7777: Connection refused (../lib/sbi/client.c:698)
07/29 18:52:07.827: [sbi] WARNING: ogs_sbi_client_handler() failed [-1] (../lib/sbi/path.c:62)
07/29 18:52:18.835: [sbi] WARNING: [e4270ac2-4dca-41ef-a8ba-f72bff6768ac] Retry registration with NRF (../lib/sbi/nf-sm.c:182)
07/29 18:52:18.835: [sbi] WARNING: [7] Failed to connect to 127.0.0.200 port 7777: Connection refused (../lib/sbi/client.c:698)
07/29 18:52:18.835: [sbi] WARNING: ogs_sbi_client_handler() failed [-1] (../lib/sbi/path.c:62)
`

-UDM-2
`Deploying component: 'udm-2'
Open5GS daemon v2.7.0-96-g322719f

07/29 18:52:07.660: [app] INFO: Configuration: '/open5gs/install/etc/open5gs/udm.yaml' (../lib/app/ogs-init.c:130)
07/29 18:52:07.660: [app] INFO: File Logging: '/open5gs/install/var/log/open5gs/udm.log' (../lib/app/ogs-init.c:133)
07/29 18:52:07.662: [sbi] INFO: NF Service [nudm-ueau] (../lib/sbi/context.c:1812)
07/29 18:52:07.662: [sbi] INFO: NF Service [nudm-uecm] (../lib/sbi/context.c:1812)
07/29 18:52:07.662: [sbi] INFO: NF Service [nudm-sdm] (../lib/sbi/context.c:1812)
07/29 18:52:07.662: [sbi] INFO: nghttp2_server() [http://127.0.0.12]:7777 (../lib/sbi/nghttp2-server.c:414)
07/29 18:52:07.663: [app] INFO: UDM initialize...done (../src/udm/app.c:31)
07/29 18:52:07.663: [sbi] WARNING: [7] Failed to connect to 127.0.0.200 port 7777: Connection refused (../lib/sbi/client.c:698)
07/29 18:52:07.663: [sbi] WARNING: ogs_sbi_client_handler() failed [-1] (../lib/sbi/path.c:62)
07/29 18:52:18.673: [sbi] WARNING: [e40de876-4dca-41ef-a67c-719e0c9ad04b] Retry registration with NRF (../lib/sbi/nf-sm.c:182)
07/29 18:52:18.674: [sbi] WARNING: [7] Failed to connect to 127.0.0.200 port 7777: Connection refused (../lib/sbi/client.c:698)
07/29 18:52:18.674: [sbi] WARNING: ogs_sbi_client_handler() failed [-1] (../lib/sbi/path.c:62)
`

Am I missing on any important detail?

Please find attached the configuration files

[my_files.zip](https://github.com/user-attachments/files/16416547/my_files.zip)


Thank you in advance.

Best regards,

**Reply by herlesupreeth** [reply]
The are no files in that zip files.

> 07/29 18:52:18.224: [dbi] WARNING: Failed to connect to server [mongodb://localhost/open5gs] (../lib/dbi/ogs-mongoc.c:122)
> 07/29 18:52:18.224: [app] ERROR: Failed to initialize PCF (../src/pcf/app.c:28)

Notice that PCF is trying to connect to mongodb on localhost. So this indicates that pcf-1_init.sh/pcf-2_init.sh is not getting executed

You could take a look at this commit for the changes required to run more than one instance of a NF - https://github.com/herlesupreeth/docker_open5gs/commit/ca0d937688c302c559a29cdfc5f4a61ef49cb335

**Reply by MerouaM** [reply]
My bad !

Please find attached the files

[files1.tar.gz](https://github.com/user-attachments/files/16430701/files1.tar.gz)

Where can I indicate that pcf-1_init.sh and pcf-2_init.sh get executed ?

Thank you for the commit that you shared with me :)

**Reply by herlesupreeth** [reply]
> Where can I indicate that pcf-1_init.sh and pcf-2_init.sh get executed ?

in the base/open5gs_init.sh

Also, following is wrong, open5gs will look for a file named pcf.yaml and not pcf-1.yaml

```
sed -i 's|PCF_IP|'$PCF_IP'|g' install/etc/open5gs/pcf-1.yaml
sed -i 's|SCP_IP|'$SCP_IP'|g' install/etc/open5gs/pcf-1.yaml
sed -i 's|NRF_IP|'$NRF_IP'|g' install/etc/open5gs/pcf-1.yaml
sed -i 's|MONGO_IP|'$MONGO_IP'|g' install/etc/open5gs/pcf-1.yaml
sed -i 's|MAX_NUM_UE|'$MAX_NUM_UE'|g' install/etc/open5gs/pcf-1.yaml
```

so in pcf-1_init.sh it should be as follows:

export DB_URI="mongodb://${MONGO_IP}/open5gs"

```
cp /mnt/pcf-1/pcf-1.yaml install/etc/open5gs/pcf.yaml
sed -i 's|PCF_IP|'$PCF_IP'|g' install/etc/open5gs/pcf.yaml
sed -i 's|SCP_IP|'$SCP_IP'|g' install/etc/open5gs/pcf.yaml
sed -i 's|NRF_IP|'$NRF_IP'|g' install/etc/open5gs/pcf.yaml
sed -i 's|MONGO_IP|'$MONGO_IP'|g' install/etc/open5gs/pcf.yaml
sed -i 's|MAX_NUM_UE|'$MAX_NUM_UE'|g' install/etc/open5gs/pcf.yaml
```

or you can change in base/open5gs_init.sh to use install/etc/open5gs/pcf-1.yaml by providing the configuration file as an argument

**Reply by MerouaM** [reply]
Hello!

I edited my files following your recommendation and the steps in the commit that you shared with me. However, I am facing another error in pcf1, pcf2, udm1, udm2. The error is the following :

```
udm1     | Deploying component: 'udm1-1'
udm1     | Open5GS daemon v2.7.0-96-g322719f
udm1     | 
udm1     | 07/31 12:24:22.982: [app] INFO: Configuration: '/open5gs/install/etc/open5gs/udm.yaml' (../lib/app/ogs-init.c:130)
udm1     | 07/31 12:24:22.982: [app] INFO: File Logging: '/open5gs/install/var/log/open5gs/udm.log' (../lib/app/ogs-init.c:133)
udm1     | 07/31 12:24:22.984: [sbi] ERROR: No udm.sbi.address: in '/open5gs/install/etc/open5gs/udm.yaml' (../lib/sbi/context.c:142)
udm1     | 07/31 12:24:22.984: [app] ERROR: Failed to initialize UDM (../src/udm/app.c:28)
udm1     | 07/31 12:24:22.984: [app] FATAL: Open5GS initialization failed. Aborted (../src/main.c:219)
```

The made the following changes:

- I changed the naming of my functions to: pcf1, pcf2, udm1, udm2 instead of pcf-1, pcf-2, udm-1, udm-2
- I added the pcf1.yaml, pcf2.yaml and pcf1_init.sh and pcf2_init.sh files under the PCF directory
- I added the udm1.yaml, udm2.yaml and udm1_init.sh and udm2_init.sh files under the UDM directory
- I added pcf1, pcf2, udm1, udm2 to the open5gs_init.sh
- I edded PCF2_IP and UDM2_IP to the .env file

Please find attached all the files that I have edited
[files3.tar.gz](https://github.com/user-attachments/files/16441916/files3.tar.gz)

Best regards :)

**Reply by herlesupreeth** [reply]
Thats strange. I didnt find any issue with the configuration files.

Did you re-compile the open5gs based image after making changes in `base` folder? If not, please recompile

And, if you still face issues then please send me the configuration file output

```
docker exec -ti udm1 bash
cat /open5gs/install/etc/open5gs/udm.yaml
```

**Comment by MerouaM** [comment]
Thank you for your reply :)

I apply build every time.

- I renamed udm1 as udm and pcf1 as pcf
- I rebuilt the image

Please find attached the configuration files.

[config-files.tar.gz](https://github.com/user-attachments/files/16480737/config-files.tar.gz)


All containers get up, apart from pcf2 and udm2 which exit with a 255 code 

`Deploying component: 'udm2-1'
Open5GS daemon v2.7.0-96-g322719f

08/03 15:24:06.105: [app] INFO: Configuration: '/open5gs/install/etc/open5gs/udm.yaml' (../lib/app/ogs-init.c:130)
08/03 15:24:06.105: [app] INFO: File Logging: '/open5gs/install/var/log/open5gs/udm.log' (../lib/app/ogs-init.c:133)
08/03 15:24:06.108: [sbi] ERROR: No udm.sbi.address: in '/open5gs/install/etc/open5gs/udm.yaml' (../lib/sbi/context.c:142)
08/03 15:24:06.108: [app] ERROR: Failed to initialize UDM (../src/udm/app.c:28)
08/03 15:24:06.108: [app] FATAL: Open5GS initialization failed. Aborted (../src/main.c:219)
`
`Deploying component: 'pcf2-1'
Open5GS daemon v2.7.0-96-g322719f

08/03 15:24:16.596: [app] INFO: Configuration: '/open5gs/install/etc/open5gs/pcf.yaml' (../lib/app/ogs-init.c:130)
08/03 15:24:16.596: [app] INFO: File Logging: '/open5gs/install/var/log/open5gs/pcf.log' (../lib/app/ogs-init.c:133)
08/03 15:24:16.598: [sbi] ERROR: No pcf.sbi.address: in '/open5gs/install/etc/open5gs/pcf.yaml' (../lib/sbi/context.c:142)
08/03 15:24:16.598: [app] ERROR: Failed to initialize PCF (../src/pcf/app.c:28)
08/03 15:24:16.598: [app] FATAL: Open5GS initialization failed. Aborted (../src/main.c:219)
`

As for the command outputs :

`docker exec -ti udm bash
cat /open5gs/install/etc/open5gs/udm.yaml`

`logger:
    file: /open5gs/install/var/log/open5gs/udm.log

sbi:
    server:
      no_tls: true
    client:
      no_tls: true

global:
  max:
    ue: 1024

udm:
    hnet:
      - id: 1
        scheme: 1
        key: /open5gs/install/etc/open5gs/hnet/curve25519-1.key
      - id: 2
        scheme: 2
        key: /open5gs/install/etc/open5gs/hnet/secp256r1-2.key
    sbi:
      server:
        - address: 172.22.0.13
          port: 7777
      service_name:
        - nudm-sdm
        - nudm-uecm
        - nudm-ee
        - nudm-pp
        - nudm-niddau
        - nudm-mt
      client:
        nrf:
          - uri: http://172.22.0.12:7777
        scp:
          - uri: http://172.22.0.35:7777
`

**Reply by herlesupreeth** [reply]
> sed -i 's|UDM_IP|'$UDM2_IP'|g' install/etc/open5gs/udm.yaml

This is wrong in udm2_init.sh since you have the following in udm2.yaml. The sed command is basically looking for string with `UDM_IP` to replace with the environment variable `UDM2_IP`  value, but in your udm2 .yaml you have string UDM2_IP so it not replaced. So, change the sed as follows `sed -i 's|UDM2_IP|'$UDM2_IP'|g' install/etc/open5gs/udm.yaml`

> 
>     sbi:
>       server:
>         - address: UDM2_IP
>           port: 7777

**Reply by MerouaM** [reply]
Thank you @herlesupreeth for your assistance.

It works now :)
